test/cloudtest/test_k8s_node_recovery.py

```diff
@@ -0,0 +1,354 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+import subprocess
+from dataclasses import dataclass
+from textwrap import dedent
+
+import pytest
+
+from materialize import ui
+from materialize.cloudtest.app.materialize_application import MaterializeApplication
+
+
+@dataclass
+class ReplicaDefinition:
+    cluster_name: str
+    index: int
+    availability_zone: str
+
+    def get_name(self) -> str:
+        return f"{self.cluster_name}_r_{self.index}"
+
+
+@dataclass
+class ClusterDefinition:
+    name: str
+    replica_definitions: list[ReplicaDefinition]
+
+    def create_replica_definitions_sql(self) -> str:
+        replica_definitions = []
+        for replica in self.replica_definitions:
+            replica_definitions.append(
+                f"{replica.get_name()} (SIZE = '1', AVAILABILITY ZONE '{replica.availability_zone}')"
+            )
+
+        return ", ".join(replica_definitions)
+
+
+def default_compute_cluster() -> ClusterDefinition:
+    compute_cluster = ClusterDefinition("c_compute", [])
+    compute_cluster.replica_definitions.append(
+        ReplicaDefinition(compute_cluster.name, 1, "1")
```

Comment:
Consider explicitly naming the last 2 arguments to ReplicaDefinition here, otherwise it is difficult to know which is which.